lib/publiccloud/instance.pm

```diff
@@ -481,6 +481,9 @@ sub softreboot {
     if ($tunneled) {
         select_console('tunnel-console', await_console => 0);
         ssh_interactive_leave();
+        while (script_run(sprintf('ssh -O check %s@%s', $args{username}, $self->public_ip)) == 0) {
```

Comment:
Is a loop really needed here? Wouldn't this do the same?

```
script_run(sprintf('ssh -O exit %s@%s || true', $args{username}, $self->public_ip))
```

Reply:
I see. In that case I would add a sleep after the `-O exit` call to avoid unnecessary load on the system.

Reply:
Done now.